"""Handled foreign key errors by changing buyer to buyers in user

Revision ID: 9945fac7da57
Revises: 7ef0ce2e2592
Create Date: 2024-05-17 13:42:44.677828

"""
from alembic import op
import sqlalchemy as sa


# revision identifiers, used by Alembic.
revision = '9945fac7da57'
down_revision = '7ef0ce2e2592'
branch_labels = None
depends_on = None


def upgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    with op.batch_alter_table('order', schema=None) as batch_op:
        batch_op.add_column(sa.Column('buyers_id', sa.Integer(), nullable=True))
        batch_op.drop_constraint('fk_order_buyer_id_user', type_='foreignkey')
        batch_op.create_foreign_key(batch_op.f('fk_order_buyers_id_user'), 'user', ['buyers_id'], ['id'])
        batch_op.drop_column('buyer_id')

    # ### end Alembic commands ###


def downgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    with op.batch_alter_table('order', schema=None) as batch_op:
        batch_op.add_column(sa.Column('buyer_id', sa.INTEGER(), nullable=True))
        batch_op.drop_constraint(batch_op.f('fk_order_buyers_id_user'), type_='foreignkey')
        batch_op.create_foreign_key('fk_order_buyer_id_user', 'user', ['buyer_id'], ['id'])
        batch_op.drop_column('buyers_id')

    # ### end Alembic commands ###